solver/source.go

```diff
@@ -21,7 +22,8 @@ type sourceOp struct {
 	src source.SourceInstance
 }
 
-func newSourceOp(_ Vertex, op *pb.Op_Source, sm *source.Manager) (Op, error) {
+func newSourceOp(_ Vertex, op *pb.Op_Source, w *worker.Worker) (Op, error) {
+	sm := w.SourceManager
```

Comment:
Maybe avoid accidental dependencies. If the worker becomes interface this can depend on a smaller interface.